app/keepers.go

```diff
@@ -162,10 +163,17 @@ func InitStakingKeeper(appCodec codec.Codec, keys map[string]*sdk.KVStoreKey, ke
 	return &stakingK
 }
 
-func initWasmKeeper(encodingConfig axelarParams.EncodingConfig, keys map[string]*sdk.KVStoreKey, keepers *KeeperCache, bApp *bam.BaseApp, appOpts types.AppOptions, wasmOpts []wasm.Option, homePath string) *wasm.Keeper {
-	wasmDir := filepath.Join(homePath, "wasm")
+func initWasmKeeper(encodingConfig axelarParams.EncodingConfig, keys map[string]*sdk.KVStoreKey, keepers *KeeperCache, bApp *bam.BaseApp, appOpts types.AppOptions, wasmOpts []wasm.Option, homePath, wasmDir string) *wasm.Keeper {
 	wasmConfig := mustReadWasmConfig(appOpts)
 
+	if wasmDir == "" {
+		dbDir := cast.ToString(appOpts.Get("db_dir"))
+		wasmDir = filepath.Join(dbDir, "wasm")
+		if dbDir == "data" {
+			wasmDir = filepath.Join(homePath, wasmDir)
+		}
```

Comment:
```suggestion
		wasmDir = filepath.Join(homePath, dbDir, "wasm")
```
`homePath`/`rootDir` is always [prefixed](https://github.com/cosmos/cosmos-sdk/blob/9abd946ba0cdc6d0e708bf862b2ca202b13f2d7b/server/util.go#L378) to the `dbDir`

Reply:
I don't think this is simplification is safe. Imagine that `dbDir` is set to an absolute path. By concatenating `homePath` to the beginning, we create an invalid path.